### ISSUE TITLE
SQL parsing performance issue for long statements

### ISSUE THREAD

**Issue by BTSmolders** [Pamblam/mysql-import#47]
Hi,

We try to update this package on our project to version 5, however, we ran into some performance issues while importing a sql file on this version.
The sql dump itself is 131kB and contains a few very long statements containing ~55000 characters.
We currently use 4.0.24 and the import of the aforementioned sql dump takes about 3-4 seconds.
When we upgrade to 5.0.21 the same file now takes 60 seconds to import.

I forked this repo and debugged this issue, and I'd like to propose a solution in this PR #46

**Comment by Pamblam** [comment]
Thank you for taking a look at it. I'm guessing the delimiter is probably not what's causing the performance issues, it's more likely going to be the process callbacks. Can you share your import code?

**Comment by BTSmolders** [comment]
I attached a zip with the test code to reproduce my claims and a test.sql from the repo and modified it with a single long statement of lorem ipsum text.
The delimiter test itself prints the import duration to the console.
When I run this test on my branch it shows that the import takes 3-4 seconds.
When I run the same test on the master branch it takes 45 secconds.

Therefore I'm fairly confident that this performance issue lies primarily in the delimiter parsing section since that is the only code I changed. 

Regarding your comment on the PR; I also updated the test.sql file with two more lines: 
```
INSERT INTO `importtest` (`id`,`doc`,`classid`) VALUES (3,'Delimiter in quoted context should be ignored ;, also in double quoted context ";" ... ',NULL);
INSERT INTO `importtest` (`id`,`doc`,`classid`) VALUES (4,'DELIMITER &&\n should not change the current delimiter because we are in a quoted context',NULL);
```

where the delimiter itself is added in a quoted context (just as in the example statement you provided in the comment), and also a `DELIMITER &&` part to make sure those texts within quotes are ignored by my modified version of the `checkNewDelimiter` function. When I run all the tests they all pass except for the '5 Rows With Semicolons Imported Into Test DB'-test which now finds 6 rows with semicolons and all 978 rows are correctly inserted into the table.

The following line in the parser makes sure that these characters are not mistaken for a DELIMITER-sql statement, and therefore I argue that the current implementation is not simply a `string.split(delimiter)`.
https://github.com/Pamblam/mysql-import/pull/46/files#diff-518958ffc2ceb664f2faff05e75b093375c1d9721ae704ff0c412a9c2691057bR520

I am looking forward to your response and whether you are able to reproduce this with the provided test files.

[test.zip](https://github.com/Pamblam/mysql-import/files/7261536/test.zip)
